bot/exts/utilities/epoch.py

```diff
@@ -0,0 +1,82 @@
+import discord
+from discord.ext import commands
+from discord.ext.commands import Converter
+from bot.bot import Bot
+from typing import Union
+
+import dateutil
+from dateutil import parser
+import arrow
+
+# https://discord.com/developers/docs/reference#message-formatting-timestamp-styles
+TIMESTAMP_FORMATS = ["H:mm A", "H:mm:ss A", "MM/DD/YYYY", "MMMM D, YYYY", "MMMM D, YYYY H:mm A",
+                     "dddd, MMMM D, YYYY hh:mm A "]
+STYLES = {"Epoch": "", "Short Time": "t", "Long Time": "T", "Short Date": "d", "Long Date": "D", "Short Date/Time": "f",
+          "Long Date/Time": "F", "Relative Time": "R"}
+
+
+class RelativeDate(Converter):
+    async def convert(self, ctx: commands.Context, argument: str) -> arrow.Arrow:
+        return arrow.utcnow().dehumanize(argument)
+
+
+class AbsoluteDate(Converter):
+    async def convert(self, ctx: commands.Context, argument: str) -> arrow.Arrow:
+        return arrow.get(dateutil.parser.parse(argument))
+
+
+class Epoch(commands.Cog):
+
+    @commands.command(name="epoch")
+    async def epoch(self, ctx: commands.Context, *, date_time: Union[RelativeDate, AbsoluteDate]) -> None:
```

Comment:
Imo, we should allow to run `.epoch` and for it return the current time. Not sure how we would add this - I guess some sort of default parameter?

Reply:
```suggestion
    async def epoch(self, ctx: commands.Context, *, date_time: Union[RelativeDate, AbsoluteDate] = arrow.utcnow()) -> None:
```
That could be done like this. 

I'm not sure that's super helpful though because you most likely will never need the current time as an epoch. I would be more inclined to send an embed with the allowed format of dates and times that you can enter in as well as timezones etc. For example info about how for absolute dates, they must go in descending order of magnitude

Reply:
True